ush/python/pygfs/task/land_analysis.py

```diff
@@ -34,6 +34,7 @@ def __init__(self, config):
 
         _res = int(self.config['CASE'][1:])
         _window_begin = add_to_datetime(self.runtime_config.current_cycle, -to_timedelta(f"{self.config['assim_freq']}H") / 2)
+        _window_begin = add_to_datetime(_window_begin, -to_timedelta(f"1S"))
```

Comment:
to do this "correctly, now don't we need to make the window_length 6hours + 1 second?

Reply:
@CoryMartin-NOAA @barlage When I conducted the snow DA cycling run with window length `6 hours + 2 seconds` which make the DA window centering at the cycle time, all the `nan` values were gone. Therefore, the DA window centering at the cycle time is important.

Reply:
CORRECTIONS: 

My above statements are incorrect, because there are still `nan` values when the above setting with the DA window beginning at 1 second back and window length 6 hours + 2 seconds. 

The above with no nan values came from the experiment with the DA window beginning at 2 second back and window length 6 hours + 1 seconds.

Reply:
Since @CoryMartin-NOAA questioned in the meeting earlier, we looked at an increment file and not all of the grids are NaN, only about 200 and they look to be in locations where there are snow observations.

Reply:
@jiaruidong2017 after a quick look, I'm not seeing any code that limits the increment, so I'm not understanding why the analysis doesn't have bad values.

Reply:
Thanks @barlage for checking the add_increment  code. I am thinking both `slmsk` and `vtype` were listed at state variables in the yaml file as `state variables: [snodl,vtype,slmsk]`. I am guessing there may be some changes made to them at somewhere in JEDI from 0/1 (water/land) value to the missing value (ice?). I need to conduct more tests for this. Please let me know if you have any suggestions.

Reply:
@CoryMartin-NOAA @barlage The `nan` values from the previous runs result from the missing bugfix in JEDI fv3-bundle (https://github.com/JCSDA-internal/fv3-jedi/pull/980) when building the GDASApp with `feature/stable-nightly` branch. 

After I separately rebuilt the GDASApp with `develop` branch including the bugfix from JEDI fv3-bundle, the following experiments were conducted including: 
**(1) 1S0S:** 
window begin: '2021-10-31T20:59:59Z'
window end: '2021-11-01T02:59:59Z'
**(2) 1S1S**
window begin: '2021-10-31T20:59:59Z'
window end: '2021-11-01T03:00:00Z'
**(3) 1S2S** (centered at `2021-11-01T00:00:00Z`)
window begin: '2021-10-31T20:59:59Z'
window end: '2021-11-01T03:00:01Z'

```
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile1.nc" and "rundir/landanl.2021110100_1S1S/anl/landinc.20211101.000000.sfc_data.tile1.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile2.nc" and "rundir/landanl.2021110100_1S1S/anl/landinc.20211101.000000.sfc_data.tile2.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile3.nc" and "rundir/landanl.2021110100_1S1S/anl/landinc.20211101.000000.sfc_data.tile3.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile4.nc" and "rundir/landanl.2021110100_1S1S/anl/landinc.20211101.000000.sfc_data.tile4.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile5.nc" and "rundir/landanl.2021110100_1S1S/anl/landinc.20211101.000000.sfc_data.tile5.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile6.nc" and "rundir/landanl.2021110100_1S1S/anl/landinc.20211101.000000.sfc_data.tile6.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile1.nc" and "rundir/landanl.2021110100_1S0S/anl/landinc.20211101.000000.sfc_data.tile1.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile2.nc" and "rundir/landanl.2021110100_1S0S/anl/landinc.20211101.000000.sfc_data.tile2.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile3.nc" and "rundir/landanl.2021110100_1S0S/anl/landinc.20211101.000000.sfc_data.tile3.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile4.nc" and "rundir/landanl.2021110100_1S0S/anl/landinc.20211101.000000.sfc_data.tile4.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile5.nc" and "rundir/landanl.2021110100_1S0S/anl/landinc.20211101.000000.sfc_data.tile5.nc" are identical.
Files "rundir/landanl.2021110100_1S2S/anl/landinc.20211101.000000.sfc_data.tile6.nc" and "rundir/landanl.2021110100_1S0S/anl/landinc.20211101.000000.sfc_data.tile6.nc" are identical.
```
All three experiments have the identical increment values regardless the setup of the window end.

Reply:
ah! great! We have a new feature/stable-nightly commit overnight, can we use that commit hash and add it in for this PR?

Reply:
Thanks Cory, and I will update the commit hash and include it in this PR.

Reply:
@CoryMartin-NOAA Okay, I will change the window length to 6hr 1sec.

Reply:
I don't understand, if the results are the same for 6hr, 6hr-1s, 6hr+1s, then why make the change?
Also, shouldn't this be handled in the observation processing part rather than in the workflow?
Consider the case where the observations are in a single file for 24h (for e.g.), shouldn't think be teased out in the yaml?

Reply:
@aerorahul this is because the 'JEDI' window convention is the opposite of the 'NCEP' window convention, and we lost that battle with Yannick a while ago. While it may be identical in this case, it shouldn't be. Observations at exactly 03z would not make it into the 06z analysis without this change. I don't quite understand you comment in the second sentence, this is what defines the window variable that goes into the templated YAML.

Reply:
I undestand.  Lets discuss offline.

Reply:
Thanks @CoryMartin-NOAA for the detailed explanations. 
@aerorahul Without this change, about several hundreds of observations will be lost in each cycle. For example, the observations increased from 1718 to 2161 with this changes.